gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -530,10 +612,50 @@ public <T> T getOption(Key<T> key) {
     return options.getOption(key);
   }
 
+  /**
+   * Validate the Universe Domain to ensure that the user configured Universe Domain and the
+   * Credentials' Universe Domain match. An exception will be raised if there are any issues when
+   * trying to validate (i.e. unable to access the universe domain).
+   *
+   * @throws UnauthenticatedException Thrown if the universe domain that the user configured does
+   *     not match the Credential's universe domain.
+   * @throws UnavailableException If client library is unable to retrieve the universe domain from
+   *     the Credentials and the RPC is configured to retry Unavailable exceptions, the client
+   *     library will attempt to retry with the RPC's defined retry bounds. If the retry bounds have
+   *     been exceeded and the library is still unable to retrieve the universe domain, the
+   *     exception will be thrown back to the user.
+   */
+  @InternalApi
+  public void validateUniverseDomain() {
+    EndpointContext endpointContext = getEndpointContext();
+    try {
+      endpointContext.validateUniverseDomain(getCredentials(), UNAUTHENTICATED_STATUS_CODE);
+    } catch (IOException e) {
+      // Check if it is an Auth Exception (All instances of IOException from endpointContext's
+      // `validateUniverseDomain()` call should be an Auth Exception).
+      if (e instanceof Retryable) {
+        Retryable retryable = (Retryable) e;
+        // Keep the behavior the same as gRPC-Java. Mark as Auth Exceptions as Unavailable
+        throw ApiExceptionFactory.createException(
+            "Unable to retrieve the Universe Domain from the Credentials. Retrying the attempt.",
```

Comment:
I don't think we want to mention `Retrying the attempt` as `retryable.isRetryable()` is not guaranteed to be true, and `UNAVAILABLE` may not be configured to be retryable by customers. I think this exception and the one below can share the same message, the only difference is the value of `retryable`.

Reply:
Makes sense. I used the same message for both and updated the message to be:
`"Unable to retrieve the Universe Domain from the Credentials."`